black.py

```diff
@@ -190,6 +191,25 @@ def supports_feature(target_versions: Set[TargetVersion], feature: Feature) -> b
     return all(feature in VERSION_TO_FEATURES[version] for version in target_versions)
 
 
+def merge_exclude(ctx: click.Context, param: click.Parameter, value: str) -> str:
+    """Injects .gitignore in excluded files"""
+    root = find_project_root(ctx.params.get("src", ()))
+    # nothing to do if no .gitignore
+    gitignore = root / ".gitignore"
+    if not gitignore.is_file():
+        return value
+    # parse gitignore
+    excluded_in_gitignore = []
+    for line in gitignore.read_text().splitlines():
+        line = line.strip().strip("/")
+        if line and not line.startswith("#"):
+            excluded_in_gitignore.append(  # noqa  W605 incorrect
+                fnmatch.translate(line).lstrip("(?s:").rstrip(r")\Z")
```

Comment:
I see what you're doing here but the required surgery on the resulting regular expression sucks a bit. Removing \Z is necessary but won't (?s:...) be harmless?